spark-unified/src/main/java/org/apache/spark/sql/delta/catalog/DeltaCatalog.java

```diff
@@ -51,4 +61,42 @@
  * </ul>
  */
 public class DeltaCatalog extends AbstractDeltaCatalog {
+
+  @Override
+  public Table newDeltaCatalogBasedTable(Identifier ident, CatalogTable catalogTable) {
+    return createBasedOnDsv2Mode(
+        () -> new SparkTable(ident, catalogTable, new HashMap<>()),
+        () -> super.newDeltaCatalogBasedTable(ident, catalogTable));
+  }
+
+  @Override
+  public Table newDeltaPathTable(Identifier ident) {
+    return createBasedOnDsv2Mode(
+        () -> new SparkTable(ident, ident.name()),
```

Comment:
why do you need to pass ident stuff twice?

Reply:
Added a comment. For path based table, spark encode the path in its identifier like delta.`/path/to/table`